pkg/cnab/provider/runtime.go

```diff
@@ -42,5 +52,28 @@ func (d *Runtime) newDriver(driverName string, claimName string, args ActionArgu
 		configurable.SetConfig(driverCfg)
 	}
 
-	return driverImpl, err
+	return driverImpl, nil
+}
+
+// If Dind support is enabled ("Docker-in-Docker"), set
+// the equivalent of the following "docker run" options:
+// 1.) -v /var/run/docker.sock:/var/run/docker.sock
+// 2.) --privileged
+func dockerDindDriver() driver.Driver {
+	dockerDriver := &docker.Driver{}
+	dockerDriver.AddConfigurationOptions(func(cfg *container.Config, hostCfg *container.HostConfig) error {
+		dockerSockMount := mount.Mount{
+			Source:   "/var/run/docker.sock",
+			Target:   "/var/run/docker.sock",
+			Type:     "bind",
+			ReadOnly: false,
+		}
+		if hostCfg.Mounts == nil {
+			hostCfg.Mounts = []mount.Mount{}
+		}
+		hostCfg.Mounts = append(hostCfg.Mounts, dockerSockMount)
+		hostCfg.Privileged = true
```

Comment:
FWIW, one item I've learned whilst configuring DinD or DooD in/for Brigade is this:

* Privileged is necessary (or alternate security setup that you'd mentioned prior) for running DinD, that is, running the docker daemon _inside_ of the container process.  (Therefore, image needs to be based on `docker:stable-dind` and/or can execute `dockerd-entrypoint.sh`)
* Mounting of the docker socket is necessary for running DooD, that is, Docker _outside_ of Docker via the host's docker socket.  No intra-container docker daemon process need be started, as it is using the host's.

See also https://docs.brigade.sh/topics/scripting/#docker-runtime and https://github.com/brigadecore/brigade/issues/1049#issuecomment-577157659

We might think about adding toggles for either/or (and both _could_ be toggled on, if for some reason this was desirable).

Reply:
Agreed, thanks for the good explanation Vaughn!

As far as the porter config goes, I think for now we can keep it as a single flag (i.e. do we want to give the bundle access to the host which both privileged and the docker sock provide). Later when we work on bundle level config, we will need to think about what the bundle can request and how Porter can creatively be providing given various environments it's installed on.